sweep/sweeper_test.go

```diff
@@ -2308,3 +2310,107 @@ func TestUpdateSweeperInputs(t *testing.T) {
 	// Assert the sweeper inputs are as expected.
 	require.Equal(expectedInputs, s.pendingInputs)
 }
+
+// TestAttachAvailableRBFInfo checks that the RBF info is attached to the
+// pending input, along with the state being marked as published, when this
+// input can be found both in mempool and the sweeper store.
+func TestAttachAvailableRBFInfo(t *testing.T) {
+	t.Parallel()
+
+	require := require.New(t)
+
+	// Create a test outpoint.
+	op := wire.OutPoint{Index: 1}
+
+	// Create a mock input.
+	testInput := &input.MockInput{}
+	testInput.On("OutPoint").Return(&op)
+	pi := &pendingInput{
+		Input: testInput,
+		state: StateInit,
+	}
+
+	// Create a mock mempool watcher and a mock sweeper store.
+	mockMempool := chainntnfs.NewMockMempoolWatcher()
+	mockStore := NewMockSweeperStore()
+
+	// Create a mempool spend event to be returned by the mempool watcher.
+	spendChan := make(chan *chainntnfs.SpendDetail, 1)
+	spendEvent := &chainntnfs.MempoolSpendEvent{
+		Spend: spendChan,
+	}
+
+	// Mock the cancel subscription calls.
+	mockMempool.On("CancelMempoolSpendEvent", spendEvent)
```

Comment:
Nit: Once()?

Reply:
fixed in a later commit, where this call is replaced with `LookupInputMempoolSpend`.